core/src/unprocessed_packet_batches.rs

```diff
@@ -61,21 +83,29 @@ impl DeserializedPacket {
             });
             Some(Self {
                 immutable_section: Rc::new(ImmutableDeserializedPacket {
-                    original_packet: packet,
+                    packet_index: *packet_index,
                     versioned_transaction,
                     message_hash,
                     is_simple_vote,
                     fee_per_cu,
                 }),
                 forwarded: false,
+                owner,
             })
         } else {
             None
         }
     }
 
+    /* TODO - to figure out how to return &Packet from borrowed RefCell
     pub fn original_packet(&self) -> &Packet {
-        &self.immutable_section.original_packet
+        let deserialized_packet_batch = self.owner.upgrade().unwrap();
+        &deserialized_packet_batch.borrow().original_packet_batch.packets[self.immutable_section.packet_index]
```

Comment:
Have not figured out how to return `&Packet` from `RefCell` wrapped object. Advices please.